pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -456,6 +580,63 @@ func (r *ReconcileRemoteMachineSet) generateMachineSets(
 	return generatedMachineSets, true, nil
 }
 
+func (r *ReconcileMachinePool) generateAWSMachineTemplates(pool *hivev1.MachinePool, cd *hivev1.ClusterDeployment, masterMachine *machineapi.Machine, logger log.FieldLogger) ([]*capiaws.AWSMachineTemplate, bool, error) {
+	if pool.DeletionTimestamp != nil {
+		return nil, true, nil
+	}
+
+	dummyMachineSets := machineapi.MachineSetList{}
+	actuator, err := r.actuatorBuilder(cd, pool, masterMachine, dummyMachineSets.Items, logger)
```

Comment:
do you mind expanding why the dummyMachineSets is required here in comments maybe?